autocompleter/base.py

```diff
@@ -222,11 +264,26 @@ def store(self, delete_old=True):
         norm_terms = self.__class__._get_norm_terms(terms)
         score = self._get_score()
         data = self.get_data()
+        facets = self.get_facets()
+
+        # Get all the facet values from the data dict
+        facet_dicts = []
+        for facet in facets:
+            try:
+                facet_value = data[facet]
+                facet_dicts.append({'key': facet, 'value': facet_value})
+            except KeyError:
+                continue
 
         old_terms = self.__class__.get_old_terms(obj_id)
-        # if old terms are the same as the new, short circuit and just
-        # update the data payload in case anything there is different.
-        if terms == old_terms:
+        old_facets = self.__class__.get_old_facets(obj_id)
+
+        updated_terms = terms != old_terms
```

Comment:
updated_terms sounds like a list of updated terms ... let's do terms_updated and facets_updated